diff --git a/dist/cjs/jsx/base.js b/dist/cjs/jsx/base.js
index 780856ecfa150bb5bcfe451d62ca0dd9ff15e3b9..17a9cb300c243914877156d17fb008179df93378 100644
--- a/dist/cjs/jsx/base.js
+++ b/dist/cjs/jsx/base.js
@@ -195,7 +195,7 @@ class JSXNode {
       }
     }
     if (emptyTags.includes(tag) && children.length === 0) {
-      buffer[0] += "/>";
+      buffer[0] += ">";
       return;
     }
     buffer[0] += ">";
diff --git a/dist/jsx/base.js b/dist/jsx/base.js
index b8a4ce03055209df1cf78b27076bd9bdbce3c45e..aa3c078a56b30bf723e829904ca59ed22eba3622 100644
--- a/dist/jsx/base.js
+++ b/dist/jsx/base.js
@@ -156,7 +156,7 @@ var JSXNode = class {
       }
     }
     if (emptyTags.includes(tag) && children.length === 0) {
-      buffer[0] += "/>";
+      buffer[0] += ">";
       return;
     }
     buffer[0] += ">";
