diff --git a/package.json b/package.json
index 77947985162c3902d24eae4d70b0b50d7ac26119..fdd56e01b526111b576296e09b029694dc340522 100644
--- a/package.json
+++ b/package.json
@@ -3,9 +3,15 @@
   "version": "2.3.1",
   "description": "A robust Punycode converter that fully complies to RFC 3492 and RFC 5891, and works on nearly all JavaScript platforms.",
   "homepage": "https://mths.be/punycode",
-  "main": "punycode.js",
-  "jsnext:main": "punycode.es6.js",
-  "module": "punycode.es6.js",
+  "type": "module",
+  "exports": {
+    ".": {
+      "import": "./punycode.js",
+      "types": "./punycode.d.ts"
+    }
+  },
+  "main": "./punycode.js",
+  "types": "./punycode.d.ts",
   "engines": {
     "node": ">=6"
   },
@@ -36,8 +42,8 @@
   "bugs": "https://github.com/mathiasbynens/punycode.js/issues",
   "files": [
     "LICENSE-MIT.txt",
-    "punycode.js",
-    "punycode.es6.js"
+    "punycode.d.ts",
+    "punycode.js"
   ],
   "scripts": {
     "test": "mocha tests",
@@ -47,12 +53,5 @@
     "codecov": "^3.8.3",
     "nyc": "^15.1.0",
     "mocha": "^10.2.0"
-  },
-  "jspm": {
-    "map": {
-      "./punycode.js": {
-        "node": "@node/punycode"
-      }
-    }
   }
 }
diff --git a/punycode.d.ts b/punycode.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..61cc175bb711898ad410ba3ae188e5ab6df12bb6
--- /dev/null
+++ b/punycode.d.ts
@@ -0,0 +1,22 @@
+declare function ucs2decode(string: string): number[];
+declare function ucs2encode(array: readonly number[]): string;
+
+declare function decode(input: string): string;
+declare function encode(input: string): string;
+declare function toASCII(input: string): string;
+declare function toUnicode(input: string): string;
+
+declare const punycode: {
+    readonly version: string;
+    readonly ucs2: {
+        readonly decode: typeof ucs2decode;
+        readonly encode: typeof ucs2encode;
+    };
+    readonly decode: typeof decode;
+    readonly encode: typeof encode;
+    readonly toASCII: typeof toASCII;
+    readonly toUnicode: typeof toUnicode;
+};
+
+export { ucs2decode, ucs2encode, decode, encode, toASCII, toUnicode };
+export default punycode;
diff --git a/punycode.es6.js b/punycode.es6.js
deleted file mode 100644
index dadece25b35f595d717ec52adfc94d23d3b1edda..0000000000000000000000000000000000000000
diff --git a/punycode.js b/punycode.js
index a1ef251924c81e2cf620132f7765058c690eeb6e..9aed7b024f1590eb0e26c2accb878b44133d4d9e 100644
--- a/punycode.js
+++ b/punycode.js
@@ -416,19 +416,15 @@ const toASCII = function(input) {
 /*--------------------------------------------------------------------------*/
 
 /** Define the public API */
-const punycode = {
+const punycode = Object.freeze({
 	/**
 	 * A string representing the current Punycode.js version number.
-	 * @memberOf punycode
-	 * @type String
 	 */
 	'version': '2.3.1',
 	/**
 	 * An object of methods to convert from JavaScript's internal character
 	 * representation (UCS-2) to Unicode code points, and back.
 	 * @see <https://mathiasbynens.be/notes/javascript-encoding>
-	 * @memberOf punycode
-	 * @type Object
 	 */
 	'ucs2': {
 		'decode': ucs2decode,
@@ -438,6 +434,7 @@ const punycode = {
 	'encode': encode,
 	'toASCII': toASCII,
 	'toUnicode': toUnicode
-};
+});
 
-module.exports = punycode;
+export { ucs2decode, ucs2encode, decode, encode, toASCII, toUnicode };
+export default punycode;
